### DISCUSSION TITLE
Serial communication with host PC for raspberry pico

### DISCUSSION THREAD

**Discussion by Prakhar005** [micropython/micropython#9493] [Using MicroPython] [answered]
Hello all,

I am using a Raspberry Pico and programming it using Thonny. I would now like to talk to the Pico using a host PC(Pico is connected to the host via USB cable) and send data serially back and forth between the Pico & the host. Is there a reliable way to do this? I checked micropython library and it states that the class USB_VCP can be used for serial comm but only for PyBoard and not the Pico. Is there any similar way to do this for the Pico?

Thank you for the help!

**Comment by jimmo** [comment, marked as answer]
The easiest way to do this is just with regular Python `print()` and `input()`. If you want to do something more sophisticated (i.e. non-blocking input) then you can use `sys.stdin` directly.

The other option is to use https://docs.micropython.org/en/latest/reference/pyboard.py.html#using-the-pyboard-library to remote control the device. Or take a look at https://github.com/BrianPugh/belay for a more comprehensive solution.

The USB_VCP on STM32 allows you to use the VCP independent of stdin/stdout. This is not currently possible on rp2 / Pico.

**Reply by Prakhar005** [reply]
Thank you very much for sharing the possible solutions. Since I would like to use non-blocking input, I tried to implement the ```sys.stdin``` command for sending data to the Pico. I was able to successfully send data to the pico. However, the issue I am now running into is that I am not able to send more than 7bytes at a time to the Pico.
It seems that pico interprets the incoming data as ASCII characters limited upto a value of 127. Is there a way to send larger sized data (say 32bits) using the ```sys.stdin``` . Thank you for the help!

**Reply by robert-hh** [reply]
When you want to use sys.stdin in a transparent way, you have to do two things:
- disable keyboard interrupt, with `micropython.kbd_intr(-1)`
- use sys.stdin.buffer. That one does not translate any incoming bytes. So there is sys.stdin.buffer.readline() and the other methods as well. Serial communication is per byte. So if you want to transfer 32bit, is has to be sent as 4  bytes.

**Reply by Prakhar005** [reply]
Thank you very much! I I used sys.stdin.buffer.read() and it worked exactly as I would have liked it to work and it does not translate any incoming bytes. Thank you for the help :)
